### DISCUSSION TITLE
Saucedacity 1.2 beta 1

### DISCUSSION THREAD

**Discussion by generic-pers0n** [saucedacity/saucedacity#27] [Announcements]
Just in time for Saucedacity's 1 year anniversary, 1.2 beta 1 has been released on this very day! This beta therefore celebrates that milestone and some new developments, but also a couple of announcements too. Announcements aside, this is a beta release, which means that **things listed here may not always make it for the final release**. Additionally, **there is always the possibility of last-minute severe bugs that exist in betas, so there is a risk of data lost with using beta builds** (although let's hope not, but that's always a possibility).

# 1.2 beta 1 Changelog
## New
* Added a new default theme, simply called "Default".
* We now provide Windows installers for 32-bit and 64-bit Windows.
* For the first time, **we are now providing some experimental macOS Intel builds**. GitHub Actions was fixed, and now we produce some rather glorious things!
  * There is a reason as to why they are experimental though: **We still don't have an actual Mac for testing these builds**. Therefore, we could actually be distributing broken macOS builds. If the builds don't work, please let us know.
  * We are currently investigating macOS ARM builds. If you are interested in helping, also let us know.

## Changed
* Dropped preference checking because that's definitely not needed...for now.
* Dropped some obsolete code.

## Broken
* AppImages are now buildable, but they are currently broken

# Known Issues

* CodeQL code scanning was set up and it reported...150 issues. Those will be fixed by 1.2 stable.
* #20 - The track view gets cut off at the bottom of the window. The cause is currently being investigated.
* #22 - FFmpeg exports don't work.

# Announcements

I have made a couple of announcements during the development of 1.2, including some recent ones. They are going to be listed here in case you missed them.

## Decision on wxWidgets
*Note: this is actually an older announcement preceding a much more larger one, as you will see below. This is kept here for historical reasons*

**I have decided to use the next stable version of wxWidgets, 3.2.0. Saucedacity will NOT (and has not been able to) build against wxWidgets 3.0**. Of course, we still are able to build against 3.1.x development versions, and we are not leaving them behind. However, we are looking towards wxWidgets 3.2.0 as the next stable version of wxWidgets, so we will be focusing on building Saucedacity against this version of wxWidgets.

For other platforms, this shouldn't mean much, maybe a couple of minor differences, but it shouldn't be anything big (except for older platforms). For Linux, however, this comes with an additional (but kind of hidden, I apologize) statement: **we are no longer focusing on GTK 2**. I've actually drop some code adjusting the font size in the device toolbar on GTK 2 in this beta, s

However, there is bigger fish to fry...

## Migration of Saucedacity to Qt
**Having seem the shortcomings of wxWidgets and how other open source applications (e.g. VLC and Dolphin), I have decided that moving to Qt for Saucedacity will be in the best interest of Saucedacity. Qt is more feature rich and also has better platform support as well, with some special points of interest at some of its offerings like QML. Either way, it is time we move on to something better, in the hopes to write better code and add some good features not previously possible.

<hr /><em>This discussion was created from the release <a href='https://github.com/saucedacity/saucedacity/releases/tag/v1.2b1'>Saucedacity 1.2 beta 1</a>.</em>

**Comment by RockyMM** [comment]
Hey there, 

I was curious about the Mac build. I just downloaded it, but I cannot start it. The app is simply not opening and there is no error log.

I am on Big Sur 11.6.4 (20G417), MacBook Pro (Retina, 15-inch, Mid 2014).

**Comment by generic-pers0n** [comment]
> Hey there, 
> 
> I was curious about the Mac build. I just downloaded it, but I cannot start it. The app is simply not opening and there is no error log.
> 
> I am on Big Sur 11.6.4 (20G417), MacBook Pro (Retina, 15-inch, Mid 2014).

Well, I kind of expected this...

The Mac builds are provided as is since I can't test them on a real Mac because I don't have one. Since we now know these builds don't work, I might have to pull them.

You can try to launch the Saucedacity executable itself from the terminal going into Saucedacity.app/Contents/Mac OS or something, I'm not exactly sure. See if anything happens there, but I don't think so.
